tangelo/toolboxes/molecular_computation/rdms.py

```diff
@@ -197,3 +197,79 @@ def energy_from_rdms(ferm_op, one_rdm, two_rdm):
     e = core_constant + np.sum(one_electron_integrals * one_rdm) + np.sum(two_electron_integrals * two_rdm)
 
     return e.real
+
+
+def pad_rdms_with_frozen_orbitals(sec_mol, onerdm, twordm):
+    """Function to pad the RDMs with the frozen orbitals data. It is based on
+    the pyscf.cccsd_rdm code, where we can set with_frozen=True.
+
+    Source:
+        https://github.com/pyscf/pyscf/blob/master/pyscf/cc/ccsd_rdm.py
+
+    Args:
+        sec_mol (SecondQuantizedMolecule): Self-explanatory.
+        onerdm (numpy.array): One-particle reduced density matrix (shape of
+            (N_active_mos,)*2).
+        twordm (numpy.array): Two-particle reduced density matrix (shape of
+            (N_active_mos,)*4).
+
+    Returns:
+        numpy.array: One-particle reduced density matrix (shape of
+            (N_total_mos,)*2).
+        numpy.array: Two-particle reduced density matrix (shape of
+            (N_total_mos,)*4).
+    """
+
+    if sec_mol.uhf:
+        raise NotImplementedError("The RDMs padding with an UHF mean-field is not implemented.")
+
+    from pyscf.lib import takebak_2d
```

Comment:
What motivated you to do that import here ?

Reply:
PySCF is not really needed for the other functions, and they are very useful for quantum computing in general.

Reply:
Yes but right now pyscf is a dependency of our package and I believe pretty much all other modules import it at the top. I think we should stay consistent with that, rather than disseminating the imports haphazardly for hypothetical situations and issues that have not happened yet (specifically the case where people want to use our software yet can't / won't install pyscf).